### DISCUSSION TITLE
Grid consumption import/export jumps a few watts if the BAT is in use

### DISCUSSION THREAD

**Discussion by alphabravocharly72** [slipx06/sunsynk-power-flow-card#567] [Ideas]
Hi,
Thx for the very useful card. One of the best. ;-)
So far all is working good. How can I avoid the import/export jumps of the Grid side? Just a few watts.
As you can see the export and Import is colored. If the battery delivers the power a little bit goes in the the grid and also a little bit is taking from the grid. The result is the color changes every few second. Is it possible to build in a +-  treshold for import/export eg range between -100 and +100W?
I Set the treshold in the Grid option but no sucess.
or the function is already available and I cant find it.
Thx

![grid](https://github.com/user-attachments/assets/e5f8eb7a-1eff-43ee-abc8-b5edd0b2c936)
![grid1](https://github.com/user-attachments/assets/83afe0d5-5486-4a37-a846-915c16db7c2b)

**Comment by slipx06** [comment]
Hi. At the moment it works as follows

```
let gridColour: string;
        switch (true) {
            case totalGridPower < 0:
                gridColour = gridExportColour;
                break;
            case totalGridPower >= 0 && totalGridPower <= Utils.toNum(config.grid?.off_threshold, 0):
                gridColour = noGridColour;
                break;
            default:
                gridColour = gridImportColour;
                break;
        }
```

The off_threshold has no effect when exporting but it does work for import. If you set it to 5W it would consider any value between 0 and 5 as the grid being off. I'll see if I can extend that to export as well

**Comment by slipx06** [comment]
I could change it like this

```
switch (true) {
    case totalGridPower < -Utils.toNum(config.grid?.off_threshold, 0): 
        gridColour = gridExportColour;
        break;
    case totalGridPower >= -Utils.toNum(config.grid?.off_threshold, 0) && totalGridPower <= Utils.toNum(config.grid?.off_threshold, 0):
        gridColour = noGridColour;
        break;
    default: // Import
        gridColour = gridImportColour;
        break;
}
```

**Comment by alphabravocharly72** [comment]
hi,
thx for your feedback.
I set the off_treshold (Grid Options)  to 10 but still showing some values on the import side.
It would be perfect if no export/import value is displayed, the grid icon should change to the disconnect icon.
thx for your support


![grid](https://github.com/user-attachments/assets/3afe6f48-bc8e-4bc8-b8ef-ca1d392194e0)

**Comment by slipx06** [comment]
Hi. The off threshold controls the colour of the icon. It wont zero the value. You should rather create a templates sensor that returns 0 when inside your threshold values and use that for your grid sensor.

The off threshold works in conjunction with the no grid colour. It stops the colour for changing but it will still show the actual grid power

![image](https://github.com/user-attachments/assets/982b66bf-c08f-4462-95b6-e8773d2cace0)

**Comment by alphabravocharly72** [comment]
Hi, now it works for the import. The value ist still in view but this is ok. If you can change the settings for the export as well would be great.
Thx a lot

![grid1](https://github.com/user-attachments/assets/c1bbc781-2797-414a-8c77-dcd00ed6ffd5)

**Comment by slipx06** [comment]
Added to 6.1.5

**Comment by alphabravocharly72** [comment]
hi, thx.
Updated.
But still not working. Can you guide me where I can find it to change the treshold for the export?

**Comment by slipx06** [comment]
You don't have to make any changes. Make sure you have cleared your frontend cache

**Comment by alphabravocharly72** [comment]
Working
Thx a lot
